### PR TITLE
Lazy loading integration

### PR DESCRIPTION
**Pull request by FinalAngel**
- Adds pagetree loading via ajax
- Implements cut, copy and paste functionality
- Implements filtering functionality

image

### CODE REVIEW

**Comment by cyberstar** [comment]
BE LGTM

File: cms/static/cms/js/modules/cms.pagetree.js [hunk 1]
```diff
@@ -185,6 +162,14 @@ var CMS = window.CMS || {};
             _setup: function _setup() {
                 var that = this;
                 var columns = [];
+                var obj = {
+                    language: this.options.lang.code,
+                    openNodes: []
+                };
+                var data = (this.options.filtered === 'true') ? false : true;
```
Comment (dhorelik) [inline cms/static/cms/js/modules/cms.pagetree.js]: maybe `var data = !this.options.filtered`?

File: cms/static/cms/js/modules/cms.pagetree.js [hunk 2]
```diff
@@ -247,32 +251,259 @@ var CMS = window.CMS || {};
...
+
+                // set event for cut and paste
+                this.ui.container.on(this.click, '.js-cms-tree-item-cut, .js-cms-tree-item-copy', function (e) {
+                    e.preventDefault();
+                    that.cache = that._getNodeData(that._getNodeId($(this)));
+                    if ($(this).hasClass('js-cms-tree-item-cut')) {
+                        that.cacheType = 'cut';
+                    } else {
+                        that.cacheType = 'copy';
+                    }
```
Comment (dhorelik) [inline cms/static/cms/js/modules/cms.pagetree.js]: `that.cacheType = $(this).hasClass('js-cms-tree-item-cut') ? 'cut' : 'copy'`

File: cms/static/cms/js/modules/cms.pagetree.js (hunk 1 above)
Comment (FinalAngel) [inline cms/static/cms/js/modules/cms.pagetree.js]: it is a string, so `false` would output true

**Comment by Brankub** [comment]
:+1:

**Pull request merged**